kite-data/kite-data-core/src/test/java/org/kitesdk/data/spi/TestRefinableViews.java

```diff
@@ -819,4 +819,23 @@ public void testNotPartitioned() throws Exception {
     Assert.assertNotNull("with should succeed",
         notPartitioned.with("timestamp", now));
   }
+
+  @Test
+  public void testReadySignals() {
+    final long instant = new DateTime(2013, 10, 1, 0, 0, DateTimeZone.UTC).getMillis();
+    final RefinableView<StandardEvent> withSpecificTimestamp =
+        unbounded.with("timestamp", instant);
+
+    Assert.assertFalse("Should not be ready initially", withSpecificTimestamp.isReady());
+
+    withSpecificTimestamp.signalReady();
+
+    Assert.assertTrue("Should be ready after signal", withSpecificTimestamp.isReady());
+
+    final RefinableView<StandardEvent> beforeSpecificTimestamp =
+        unbounded.toBefore("timestamp", instant);
```

Comment:
I think this should also test instant +1, which will be in the same physical partition but shouldn't be signalled.